yarl/_url.py

```diff
@@ -209,38 +210,50 @@ def __new__(
         else:
             raise TypeError("Constructor parameter should be str")
 
+        cache: Dict[str, Union[str, int, None]] = {}
         if not encoded:
             host: Union[str, None]
-            if not val[1]:  # netloc
-                netloc = ""
+            scheme, netloc, path, query, fragment = val
+            if not netloc:  # netloc
                 host = ""
             else:
-                host = val.hostname
+                username, password, host, port = cls._split_netloc(val[1])
                 if host is None:
                     raise ValueError("Invalid URL: host is required for absolute urls")
-
-                try:
-                    port = val.port
-                except ValueError as e:
-                    raise ValueError(
-                        "Invalid URL: port can't be converted to integer"
-                    ) from e
-
+                host = cls._encode_host(host)
+                raw_user = None if username is None else cls._REQUOTER(username)
+                raw_password = None if password is None else cls._REQUOTER(password)
```

Comment:
Should make sure we have an explicit test here. Not sure if an empty string is a valid password

Reply:
Already covered by `test_empty_password_without_user`